File: app/src/main/java/com/tech/dagger2_android/inJava/network/NetworkConnection.java
package com.tech.dagger2_android.inJava.network;

import com.tech.dagger2_android.BuildConfig;
import com.tech.dagger2_android.inJava.utils.Constants;

import javax.inject.Inject;

public class NetworkConnection {
    public String mEndPoint;

    @Inject
    public NetworkConnection(){
        if(BuildConfig.DEBUG){
            this.mEndPoint = Constants.DEBUG_ENDPOINT;
        }else{
            this.mEndPoint = Constants.PROD_ENDPOINT;
        }
    }
    public String doReq(){
        return this.mEndPoint;
    }
}

File: app/src/main/java/com/tech/dagger2_android/inKotlin/network/NetworkConnection.kt
package com.tech.dagger2_android.inKotlin.network

import com.tech.dagger2_android.inKotlin.utils.Constants
import javax.inject.Inject

class NetworkConnection @Inject constructor(){
    private var mEndPoint: String? = null

    init {
        if (mEndPoint == null) {
            mEndPoint = Constants.DEBUG_ENDPOINT
        } else {
            mEndPoint = Constants.PROD_ENDPOINT
        }
    }

    fun doReq(): String? {
        return mEndPoint
    }
}